### ISSUE TITLE
Allow to define alt text in MediaPickerField

### ISSUE THREAD

**Issue by jptissot** [OrchardCMS/OrchardCore#4326]
We need to be able to define localizable alt text for Media. I thought the easiest place would be in the MediaPickerField. Thoughts ?

**Comment by hishamco** [comment]
If there's ability to enter the `alt` text from the admin, the current implementation will unable to localize it, but when we add the dynamic string localization, this will be a good enhancement

**Comment by jptissot** [comment]
Best would be to allow alt text to be defined with the media files and be localizable. Maybe it already is possible. If not, I am ok with the MediaPickerField holding the alt text. 
/cc @matiasmolleja

**Comment by deanmarcussen** [comment]
@jptissot The way I've normally handled that, and I believe is the design intention, is that if we want metadata (alt tags) on a MediaField, we should wrap that field into an ImagePart, and place metadata on that, i.e. Caption on the Agency sample.

Then it's localizable right? 
but does that mean that you have to pick the image again?

**Comment by jptissot** [comment]
I guess that could work. Create a ReusablePart called ImagePart and add a MediaField to it + a Alt Text field. 

I just thought the base MediaPickerField should have at least the alt text since it's pretty common and nessesary for accessibility

**Comment by Skrypt** [comment]
I think the goal in OC is to not bind any metadata by default to the medias. It's easier to migrate medias that way. In 01 in order to be able to migrate medias you needed to have database records migrated for them to work. The goal in RC was to make media more independent of it's metadata.

What @deanmarcussen says is what needs to be documented for this specific usage.

**Comment by jptissot** [comment]
What about my idea of adding a Textbox to the MediaPickerField for alt text? The metadata would be independent from the media but be modifyable from the ContentItem edit page ?

**Comment by Skrypt** [comment]
Yeah, you should check with @matiasmolleja for this. I think we need to come up in this case with a procedure to add a Textbox to the MediaPickerField and a documentation on how to reproduce it. Because your TextField could be named "Apple" or "Oranges" ... and the Vue.js ViewComponent in that case needs to have some constants to look at to add this.

**Comment by jptissot** [comment]
I was thinking something along the lines of what we do for the LinkField. It has 2 textbox (url and text) The MediaPickerField would have the Media picker and another Textbox for alt text. I dont think we need to modify the Vue component.

**Comment by Skrypt** [comment]
Hmm ask @sebastienros what he thinks about it first.

**Comment by matiasmolleja** [comment]
Something like this has been discussed before:
#908  #3404
There are pros and cons. I would would not add more complexity to the field. I think that we should leverage the content part concept for that. It is a matter of adding docs about how to do that, and maybe add a content part or content type to one or several of our default themes.

It is true that this is a so common requirement that we should provide an easy way to achieve it.
But I will be glad to help if people agree on adding this to the field.  😄

**Comment by sebastienros** [comment]
Another option would be to have a `MediaPickerWithAltTextField` class that would be used by the editor, and the dev if they know they want to access the value. But it would still not be in the public field. And we'd need to check if we can extend graphql with this new data too.
